Sources/SymbolKit/SymbolGraph/Misc/SemanticVersionPrerelease.swift

```diff
@@ -0,0 +1,135 @@
+/*
+ This source file is part of the Swift.org open source project
+ 
+ Copyright (c) 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+ 
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+extension SymbolGraph.SemanticVersion {
+	/// A storage for pre-release identifiers.
+	internal struct Prerelease {
+		/// The identifiers.
+		internal let identifiers: [Identifier]
+		/// A pre-release identifier.
+		internal enum Identifier {
+			/// A numeric pre-release identifier.
+			/// - Parameter identifier: The identifier.
+			case numeric(_ identifier: Int)
+			/// An alphanumeric pre-release identifier.
+			/// - Parameter identifier: The identifier.
+			case alphanumeric(_ identifier: String)
+		}
+	}
+}
```

Comment:
Because the precedence between pre-release identifiers depends on whether they're numeric, their numeric-ness is recorded this way right after each identifier is validated during the version's initialization.